File: frontend/src/types/schedule.ts
export interface ScheduleSlot {
  id: string;
  lessonId: string;
  lessonTitle: string;
  weekday: number;
  weekdayLabel: string;
  time: string;
  isEnabled: boolean;
  nextAt: string | null;
  nextHint: string;
  createdBy: string;
}

export const WEEKDAY_OPTIONS = [
  { id: 0, label: "Пн", full: "Понедельник" },
  { id: 1, label: "Вт", full: "Вторник" },
  { id: 2, label: "Ср", full: "Среда" },
  { id: 3, label: "Чт", full: "Четверг" },
  { id: 4, label: "Пт", full: "Пятница" },
  { id: 5, label: "Сб", full: "Суббота" },
  { id: 6, label: "Вс", full: "Воскресенье" },
];

File: backend/app/schemas/schedule.py
"""Схемы расписания для пульта педагога."""

from pydantic import BaseModel, Field


class ScheduleCreate(BaseModel):
    """Новый слот: какой день, во сколько, какая программа."""

    lessonId: str
    weekday: int
    time: str = Field(examples=["10:00"])
    isEnabled: bool = True


class ScheduleUpdate(BaseModel):
    """Частичное обновление слота."""

    lessonId: str | None = None
    weekday: int | None = None
    time: str | None = None
    isEnabled: bool | None = None


class ScheduleTickRequest(BaseModel):
    """Принудительная проверка слотов (стенд и тесты)."""

    at: str | None = None


class SchedulePublic(BaseModel):
    """Слот в списке пульта."""

    id: str
    lessonId: str
    lessonTitle: str
    weekday: int
    weekdayLabel: str
    time: str
    isEnabled: bool
    nextAt: str | None = None
    nextHint: str
    createdBy: str


class ScheduleTickResult(BaseModel):
    """Что сделала проверка расписания."""

    started: list[str]
    skipped: str | None = None
